src/metaxy/graph/status.py

```diff
@@ -0,0 +1,132 @@
+"""Feature metadata status inspection utilities.
+
+This module provides reusable SDK functions for inspecting feature metadata status,
+useful for both CLI commands and programmatic usage.
+"""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Any
+
+import narwhals as nw
+import polars as pl
+
+if TYPE_CHECKING:
+    from metaxy.metadata_store.base import MetadataStore
+    from metaxy.models.feature import BaseFeature
+
+
+def count_lazy_rows(lazy_frame: nw.LazyFrame[Any]) -> int:
+    """Return row count for a Narwhals LazyFrame.
+
+    Args:
+        lazy_frame: The LazyFrame to count rows from
+
+    Returns:
+        Number of rows in the LazyFrame
+    """
+    return lazy_frame.select(nw.len()).collect().to_polars()["len"].item()
+
+
+def preview_samples(
+    lazy_frame: nw.LazyFrame[Any],
+    id_columns: Sequence[str] | None = None,
+    limit: int = 5,
+) -> pl.DataFrame:
+    """Return preview of samples as a Polars DataFrame.
+
+    Args:
+        lazy_frame: The LazyFrame containing samples
+        id_columns: Columns to include in the preview (defaults to ["sample_uid"])
+        limit: Maximum number of rows to include
+
+    Returns:
+        Polars DataFrame with the requested columns and row limit
+    """
+    # Determine columns to display
+    headers = list(id_columns or ["sample_uid"])
+    available_headers = [col for col in headers if col in lazy_frame.columns]
+
+    if not available_headers:
+        # Fallback to all columns if specified headers don't exist
+        available_headers = lazy_frame.columns
+
+    if not available_headers:
+        return pl.DataFrame()
+
+    # Only collect limited rows
+    return lazy_frame.select(available_headers).head(limit).collect().to_polars()
+
+
+def get_feature_metadata_status(
```

Comment:
I think this needs a parameter to allow reading from fallback stores. Should default to false.